src/app/common/hooks/analytics/transactions-analytics.hooks.ts

```diff
@@ -111,3 +118,27 @@ export function useTrackChangedTransactions(
   );
   return result;
 }
+
+function useIsFirstDeposit(stxBalance: AccountStxBalanceBigNumber | undefined): boolean {
+  const currentNetwork = useCurrentNetworkState();
+  const hadZeroStxBalance = useAnalyticsHadZeroStx()[currentNetwork.chainId];
+  const hasZeroStx = stxBalance?.total_received.isEqualTo(0) || false;
```

Comment:
This could be `useMemo`'d as it is also an operation that takes places on every render.

Reply:
This check is not expensive in computation, that's why I did not add useMemo, but I can add it.